pkg/sharing/sharing.go

```diff
@@ -252,6 +251,31 @@ func removeSharingTrigger(inst *instance.Instance, triggerID string) error {
 	return nil
 }
 
+// RevokeByNotification is called on the recipient side, after a revocation
+// performed by the sharer
+func (s *Sharing) RevokeByNotification(inst *instance.Instance) error {
+	if s.Owner {
+		return ErrInvalidSharing
+	}
+	if err := DeleteOAuthClient(inst, &s.Members[0], &s.Credentials[0]); err != nil {
+		return err
+	}
+	if !s.ReadOnly() {
```

Comment:
I think you can skip the `if !s.ReadOnly()` condition, and just rely on the sharing trigger IDs to know if there are some triggers to remove (or not).

Reply:
3 checks that are doing no I/O or memory allocation are really fast, probably even faster than calling `ReadOnly()`. But we don't care, it's a micro-optimization on a cold path, readability and solidity are way more important than speed here.